clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -183,4 +197,50 @@ public synchronized long lastSuccessfulUpdate() {
     public long refreshBackoff() {
         return refreshBackoffMs;
     }
+
+    /**
+     * Set state to indicate if metadata for all topics in Kafka cluster is required or not.
+     * @param needMetadaForAllTopics boolean indicating need for metadata of all topics in cluster.
+     */
+    public void needMetadataForAllTopics(boolean needMetadaForAllTopics) {
+        this.needMetadataForAllTopics = needMetadaForAllTopics;
```

Comment:
This might not work as well as I thought since the metadata is also updated in partitionsFor(). I actually think it would be better for partitionsFor() to use the same approach as listTopics(), but that is out of scope for this issue.